hibernate-core/src/main/java/org/hibernate/id/enhanced/HiLoOptimizer.java

```diff
@@ -53,12 +53,78 @@
 public class HiLoOptimizer extends AbstractOptimizer {
 	private static final Logger log = Logger.getLogger( HiLoOptimizer.class );
 
-	private static class GenerationState {
+	protected static class GenerationState {
+		private final int incrementSize;
```

Comment:
not sure about having the `incrementSize` here and a copy in the `AbstractOptimizer`, plus the check done in the `GenerationState` constructor
 `if ( incrementSize < 1 ) {
	throw new HibernateException( "increment size cannot be less than 1" );
}`
has been already done in the `HiLoOptimizer` one. 

What about adding an `incrementSize`  parameter to the `updateFromNewSourceValue` method